src/hex_http.erl

```diff
@@ -0,0 +1,31 @@
+-module(hex_http).
+-export([get/3]).
+-ifdef(TEST).
+-export([user_agent/1]).
+-endif.
+-include_lib("hex_erl.hrl").
+
+-type client() :: #{adapter => module(), user_agent_string => string()}.
+-type status() :: non_neg_integer().
+-type headers() :: map().
+
+-callback get(URI :: string(), headers()) ->
+    {ok, status(), headers(), binary()} |
+    {error, term()}.
+
+-spec get(client(), string(), headers()) -> {ok, {status(), headers(), binary()}} | {error, term()}.
+get(#{adapter := Adapter, user_agent_string := UserAgentString}, URI, Headers) ->
+    Headers2 = put_new("user-agent", user_agent(UserAgentString), Headers),
+    Adapter:get(URI, Headers2).
+
+user_agent(UserAgentString) ->
+    OTPRelease = erlang:system_info(otp_release),
+    ERTSVersion = erlang:system_info(version),
+    OTPString = " (OTP/" ++ OTPRelease ++ " erts-" ++ ERTSVersion ++ ")",
```

Comment:
@ericmj I've added erts version so the final UA looks like this:

```
1> hex_http:user_agent("(test)").
"hex_erl/0.1.0 (test) (OTP/20 erts-9.2.1)"
```

I remember that @fishcakez told me that current way we get OTP version (https://github.com/hexpm/hex/blob/v0.17.3/lib/hex/utils.ex#L211) on Hex is not reliable but I don't remember if I followed up on it so here we are :)

Reply:
updated to:

```
1> hex_http:user_agent("(test)").
"hex_erl/0.1.0 (test) (OTP/20) (erts/9.2.1)"
```